tests/appsec/test_processor.py

```diff
@@ -132,6 +132,40 @@ def test_headers_collection(tracer):
     assert span.get_tag("http.response.headers.foo") is None
 
 
+@pytest.mark.parametrize(
+    "payload,tag,expected_result",
+    [
+        (
+            {
+                "status_code": "200",
+                "request_cookies": {"cookie1": "im the cookie1"},
+            },
+            "http.request.cookies.cookie1",
+            None,
+        ),
+        (
+            {
+                "status_code": "200",
+                "request_cookies": {"cookie1": "im the cookie1"},
+            },
+            "http.request.cookies",
+            None,
+        ),
+    ],
+)
+def test_cookies_collection(payload, tag, expected_result, tracer):
+    _enable_appsec(tracer)
+    with tracer.trace("test", span_type=SpanTypes.WEB) as span:
+        payload["span"] = span
+        payload["integration_config"] = Config()
+        payload["raw_uri"] = "http://example.com/.git"
+        payload["status_code"] = "404"
+        set_http_meta(**payload)
+
+    # Because is sensitive data, server.request.cookies should be None
+    assert span.get_tag(tag) == expected_result
```

Comment:
Does this test need to be parametrized? are there other examples you can think of that should be added here? Are there cases when we collect the cookie tags vs not (e.g. an attack is detected in a cookie) ?

Otherwise it seems we can just hardcode the payload and change this to:

```suggestion
    # Because is sensitive data, server.request.cookies should be None
    assert span.get_tag("http.request.cookies") is None
    assert span.get_tag("http.request.cookies.cookie1") is None
```

Reply:
do we need/want to assert that calling `set_http_meta` with request cookies adds them to the context?

Reply:
The function is parametrized because It's a good practice "1 test -> 1 assert". We want to verify that cookies aren't in the context because cookies are sensitive data (To avoid mistakes as I had in my first commit :sweat_smile: )

Reply:
🤷🏻 either way is fine with me. personally I am not too worried about "1 test -> 1 assert", but no strong opinion.

Reply:
actually, we could make this a snapshot test.

keep it parametrized for the payload data, and then result is always going to be that no cookies prefixed with `http.request.cookies` is ever in the result.


Then the test would become:

```python

@pytest.mark.parametrize(
    "payload",
    [
        (
            {
                "status_code": "200",
                "request_cookies": {"cookie1": "im the cookie1"},
            },
        ),
    ],
)
@snapshot(include_tracer=True)
def test_cookies_collection(payload, tag, expected_result, tracer):
    _enable_appsec(tracer)
    with tracer.trace("test", span_type=SpanTypes.WEB) as span:
        payload["span"] = span
        payload["integration_config"] = Config()
        payload["raw_uri"] = "http://example.com/.git"
        payload["status_code"] = "404"
        set_http_meta(**payload)
```


Since the snapshot trace/span/tags should always be the same regardless of the input.